rpc/gnmi/gnmignoissh-dialout-grpctunnel.md

```diff
@@ -27,48 +27,55 @@ Carl Lebsack
+For example, a tunnel client can convey to the tunnel server the unique name of an available target (target_id) and service for that target (target_type = “GNMI_GNOI”). Both target_id and target_type are strings to allow for the most flexible tunnel deployment options.  The inclusion of a TargetType enum in the tunnel.proto enables the standardization of a set of predefined protocols to be carried by the tunnel.  However, a protocol not included within the enum can still be used within the tunnel so long as the tunnel client and tunnel server agree on a common string for the type.
 
 Specifically, the target_id is not prescribed, but should be unique in a given deployment. Some examples of what can be used for the target_id include hostname, chassis serial number or mac address, management IP address.  This target_id is assumed to be leveraged within gNMI as the [Path target field](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#2221-path-target) in the case the tunnel is hosting a gNMI interface and allows a tunnel server to redirect a gNMI request over the corresponding tunnel session. The value of this id is expected to be defined by a network operator and should be configurable on each target to be provided when dialing out to a tunnel server.
 
-The target_type field, although flexible, is used in conjunction with target_id to differentiate among multiple protocols supported for the same target.  It is a string to allow for arbitrary TCP protocols to be supported over the tunnel.  However, standard protocols are explicitly included in the tunnel Protocol enum, (e.g. GNMI_GNOI, SSH), and the string value for that enumeration can be reliably used on both tunnel clients and servers.
+The target_type field, although flexible, is used in conjunction with target_id to differentiate among multiple protocols supported for the same target.  It is a string to allow for arbitrary TCP protocols to be supported over the tunnel.  However, standard protocols are explicitly included in the tunnel Protocol enum, (e.g. GNMI_GNOI, SSH), and the string value for that enumeration can be reliably used on both tunnel clients and tunnel servers.
+
+The diagram below shows the call flow of the tunnel establishment. It uses a 
+standalone tunenl server for the purpose illustration. Alternative deployment will be discussed in more details in [Select deployment use cases](#select-deployment-use-cases). Here each network device (gNMI server or target device) will establish a gRPC tunnel with the tunnel server by finishing the registration step. After that, the network devices will communicate as if they were directly connected.
```

Comment:
s/tunenl/tunnel/
s/purpose illustration/purpose of this illustration/
s/depoloyment/deployments/
s/details/detail/
s/with the tunnel server/with one or more tunnel servers/
s/communicate as if they were directly connected/accept future connections over the created tunnels/